File: Codility/Codility-py/04-counting-elements/PermCheck.py
# PermCheck


def solution(a):
    a.sort()
    for i in range(1, len(a) + 1):
        if i != a[i - 1]:
            return 0

    return 1


# (100%) https://codility.com/demo/results/trainingX98HWU-GHV/

print(solution([4, 1, 3, 2]))  # 1
print(solution([4, 1, 3]))  # 0

File: Codility/Codility-js/4-counting-elements/PermCheck.js
function solution(A) {
    var arr = [];
    for (var i = 0; i < A.length; i++){
        if (arr[A[i]]) return 0;
        arr[A[i]] = true;
    }
    for (var i = 1; i < arr.length; i++){
        if(!arr[i]) return 0;
    }
    return 1;
}

console.log(solution([1,2,3,3,4]));